src/learned.jl

```diff
@@ -0,0 +1,30 @@
+##--- Learnable Transforms ---##
+
+type CenteringTransformation{T<:Real}
+    shift::ShiftTransform{T}
+end
+CenteringTransformation() = CenteringTransformation(ShiftTransform())
+function CenteringTransformation(x::AbstractArray)
+    tfm = CenteringTransformation()
+    learn!(tfm, x)
+end
+learn!(tfm::CenteringTransformation, x) = (tfm.shift = mean(x); tfm)
+transform!(tfm::CenteringTransformation, x) = (transform!(tfm.shift, x); x)
+invert!(tfm::CenteringTransformation, x) = (invert!(tfm.shift, x); x)
+
+type Standardize{T<:Real}
```

Comment:
I am a twitchy about `CenteringTransformation` having a postfix while `Standardize` doesn't. I think we should go one way or the other here.